commcare_connect/static/js/project.js

```diff
@@ -63,3 +63,72 @@ function addAccuracyCircles(map, visit_data) {
 }
 
 window.addAccuracyCircles = addAccuracyCircles;
+
+function addCatchmentAreas(map, catchments) {
+  const ACTIVE_COLOR = '#3366ff';
+  const INACTIVE_COLOR = '#ff4d4d';
+  const CIRCLE_OPACITY = 0.3;
+
+  map.on('load', () => {
+    const catchmentCircles = catchments.map((catchment) =>
+      circle([catchment.lng, catchment.lat], catchment.radius, {
+        units: 'meters',
+        properties: { active: catchment.active },
+      }),
+    );
+
+    map.addSource('catchment_circles', {
+      type: 'geojson',
+      data: {
+        type: 'FeatureCollection',
+        features: catchmentCircles,
+      },
+    });
+
+    map.addLayer({
+      id: 'catchment-circles-layer',
+      source: 'catchment_circles',
+      type: 'fill',
+      paint: {
+        'fill-color': ['case', ['get', 'active'], ACTIVE_COLOR, INACTIVE_COLOR],
+        'fill-opacity': CIRCLE_OPACITY,
+      },
+    });
+
+    map.addLayer({
+      id: 'catchment-circle-outlines-layer',
+      source: 'catchment_circles',
+      type: 'line',
+      paint: {
+        'line-color': '#fcbf49',
+        'line-width': 3,
+        'line-opacity': 0.5,
+      },
+    });
+
+    // Add legend
+    const legend = document.createElement('div');
+    legend.className = 'card position-absolute bottom-0 end-0 m-2';
+
+    const squareBoxStyle = 'width: 20px; height: 20px; opacity: 0.3;';
+    const activeStyle = `${squareBoxStyle} background-color: ${ACTIVE_COLOR};`;
+    const inactiveStyle = `${squareBoxStyle} background-color: ${INACTIVE_COLOR};`;
+
+    legend.innerHTML = `
+                    <div class="card-body">
+                        <h6 class="card-title">Catchment Areas</h6>
+                        <div class="mb-2 d-flex align-items-center">
+                            <span class="d-inline-block me-2" style="${activeStyle}"></span>
+                            <span>Active</span>
+                        </div>
+                        <div class="d-flex align-items-center">
+                            <span class="d-inline-block me-2" style="${inactiveStyle}"></span>
+                            <span>Inactive</span>
+                        </div>
+                    </div>
+                `;
+    map.getContainer().appendChild(legend);
```

Comment:
We generally try to avoid adding HTML via Javascript in this way (because it's hard to reason with the HTML if you have HTML scattered around various files). Have you explored if it would be possible to add this directly to the HTML and conditionally show it. We have `x-show` and `x-component` in AlpineJS that might be useful here.

Reply:
Agree with sravan on this one, thanks for updating

Reply:
@sravfeyn  and @calellowitz  Added the  changes.